ballerina/tests/fromString_test.bal

```diff
@@ -0,0 +1,280 @@
+    test:assertEquals(stringValue3, "true");
+}
+
+@test:Config
+isolated function testStringToUnion() returns error? {
+    float|decimal value1 = check fromStringWithType("1.2");
+    test:assertTrue(value1 is float);
+    test:assertEquals(value1, 1.2);
+
+    decimal|float|int value2 = check fromStringWithType("1.2");
+    test:assertTrue(value2 is float);
+    test:assertEquals(value2, 1.2);
+
+    float|decimal value3 = check fromStringWithType("1");
+    test:assertTrue(value3 is float);
+    test:assertEquals(value3, 1.0);
+
+    decimal|float|int value4 = check fromStringWithType("1");
+    test:assertTrue(value4 is int);
+    test:assertEquals(value4, 1);
+
+    ()|string value5 = check fromStringWithType("null");
+    test:assertTrue(value5 is ());
+    test:assertEquals(value5, ());
+
+    ()|string value6 = check fromStringWithType("()");
+    test:assertTrue(value6 is ());
+    test:assertEquals(value6, ());
+
+    boolean|string value7 = check fromStringWithType("true");
+    test:assertTrue(value7 is boolean);
+    test:assertEquals(value7, true);
+
+    boolean|string value8 = check fromStringWithType("false");
+    test:assertTrue(value8 is boolean);
+    test:assertEquals(value8, false);
+
+    int|boolean value9 = check fromStringWithType("1");
+    test:assertTrue(value9 is int);
+    test:assertEquals(value9, 1);
+
+    float|decimal|boolean value10 = check fromStringWithType("0");
+    test:assertTrue(value10 is float);
+    test:assertEquals(value10, 0.0);
+
+    decimal|boolean value11 = check fromStringWithType("0");
+    test:assertTrue(value11 is decimal);
+    test:assertEquals(value11, 0.0d);
+
+    int|string value12 = check fromStringWithType("1");
+    test:assertTrue(value12 is int);
+    test:assertEquals(value12, 1);
+
+    float|string value13 = check fromStringWithType("1.2");
+    test:assertTrue(value13 is float);
+    test:assertEquals(value13, 1.2);
+
+    decimal|string value14 = check fromStringWithType("1.2");
+    test:assertTrue(value14 is decimal);
+    test:assertEquals(value14, 1.2d);
+
+    int|string value15 = check fromStringWithType("abc");
+    test:assertTrue(value15 is string);
+    test:assertEquals(value15, "abc");
+
+    int|string value16 = check fromStringWithType("9223372036854775808");
+    test:assertTrue(value16 is string);
+    test:assertEquals(value16, "9223372036854775808");
+
+    int|string value17 = check fromStringWithType("-9223372036854775809");
+    test:assertTrue(value17 is string);
+    test:assertEquals(value17, "-9223372036854775809");
+
+    decimal|string value18 = check fromStringWithType("9.99E+61111");
+    test:assertTrue(value18 is string);
+    test:assertEquals(value18, "9.99E+61111");
+
+    decimal|string value19 = check fromStringWithType("-9.99E+61111");
+    test:assertTrue(value19 is string);
+    test:assertEquals(value19, "-9.99E+61111");
+}
+
+// Negative tests for fromStringWithType() function.
```

Comment:
Shall we add test cases with unsupported types for `fromStringWithType`

Reply:
Those will be a compile-time error. Right?

Reply:
I was thinking more about currently unsupported types such as in [here](https://github.com/hasithaa/module-ballerina-data/pull/3/files#r1242010450). Invalid types will give a compiler-error

Reply:
Handled the reference type cases [f3eb6dc](https://github.com/hasithaa/module-ballerina-data/pull/3/commits/f3eb6dcbdbb4f80b0ed09b184702c48565e3212d)